mesonbuild/interpreter/interpreter.py

```diff
@@ -3473,23 +3473,6 @@ def build_target(self, node: mparser.BaseNode, args: T.Tuple[str, SourcesVarargs
         self.project_args_frozen = True
         return target
 
-    def kwarg_strings_to_includedirs(self, kwargs: kwtypes._BuildTarget) -> None:
-        if kwargs['d_import_dirs']:
-            items = kwargs['d_import_dirs']
-            cleaned_items: T.List[build.IncludeDirs] = []
-            for i in items:
-                if isinstance(i, str):
-                    # BW compatibility. This was permitted so we must support it
-                    # for a few releases so people can transition to "correct"
-                    # path declarations.
-                    if os.path.normpath(i).startswith(self.environment.get_source_dir()):
-                        mlog.warning('''Building a path to the source dir is not supported. Use a relative path instead.
-This will become a hard error in the future.''', location=self.current_node)
-                        i = os.path.relpath(i, os.path.join(self.environment.get_source_dir(), self.subdir))
```

Comment:
This deprecation warning (or should be `mlog.deprecated` instead of `mlog.warning`) is getting lost. It's important, and needs to be ported to extract_incdirs. And the test that ensures this keeps working should not drop this.

Reply:
Since we already handle some feature changes in extract_incdirs, we should handle this one inside it too, but preferably using a `FeatureDeprecated` mirror the using of the `FeatureNew` inside `extract_incdirs`

Reply:
This is now implemented. Let me know what you think.